detection_rules/rule.py

```diff
@@ -745,7 +803,7 @@ def get_supported_version(self) -> str:
         min_stack = self.convert_supported_version(rule_min_stack)
         return f"{min_stack.major}.{min_stack.minor}"
 
-    def _post_dict_transform(self, obj: dict) -> dict:
+    def _post_dict_conversion(self, obj: dict) -> dict:
```

Comment:
renamed this and all related methods to differentiate.

* transform: 
   * dynamic manipulations of existing fields
   * defined in TOML [transform] and processed in BaseRuleData class `process_transforms`
* conversions: 
    * dynamic creation of new fields
    *  processed in this method